src/components/data_submission/DataAccessGovernance.js

```diff
@@ -1,18 +1,73 @@
 import ConsentGroupForm from './consent_group/ConsentGroupForm';
 import { useEffect, useState, useCallback } from 'react';
-import { isNil, every, cloneDeep } from 'lodash/fp';
+import { isNil, every, cloneDeep, isEmpty, find } from 'lodash/fp';
 import { div, h, h2, a, span } from 'react-hyperscript-helpers';
-import { DAC } from '../../libs/ajax';
+import { DAC, DAR } from '../../libs/ajax';
 
 import './ds_common.css';
 
 export const DataAccessGovernance = (props) => {
   const {
-    onChange, onFileChange, validation, onValidationChange, setAllConsentGroupsSaved
+    onChange,
+    studyEditMode,
+    onFileChange,
+    validation,
+    datasets,
+    onValidationChange,
+    setAllConsentGroupsSaved
   } = props;
 
   const [consentGroupsState, setConsentGroupsState] = useState([]);
   const [dacs, setDacs] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const getDiseaseLabels = async (ontologyIds) => {
+    let labels = [];
+    if (!isEmpty(ontologyIds)) {
+      const idList = ontologyIds.join(',');
+      const result = await DAR.searchOntologyIdList(idList);
+      labels = result.map(r => r.label);
+    }
+    return labels;
+  };
```

Comment:
This looks like a copy of `DatasetUpdate.getDiseaseLabels`. We should try to extract common functions like this. It might be helpful to include the try/catch in the normalize function below to make the function more resilient.